solidity/test/MezoAllocator.test.ts

```diff
@@ -542,32 +733,142 @@ describe("MezoAllocator", () => {
     })
 
     context("when the caller is governance", () => {
+      beforeAfterSnapshotWrapper()
+
+      context("when there is no deposit", () => {
+        beforeAfterSnapshotWrapper()
+
+        context("when there is no donation", () => {
+          beforeAfterSnapshotWrapper()
+
+          let tx: ContractTransactionResponse
+
+          before(async () => {
+            tx = await mezoAllocator.connect(governance).releaseDeposit()
+          })
+
+          it("should not emit DepositReleased event", async () => {
+            await expect(tx).to.not.emit(mezoAllocator, "DepositReleased")
+          })
+
+          it("should not call MezoPortal.withdraw function", async () => {
+            await expect(tx).to.not.emit(mezoPortal, "WithdrawFully")
+          })
+        })
+
+        context("when there is a donation", () => {
+          beforeAfterSnapshotWrapper()
+
+          const donationAmount = to1e18(2)
+
+          let tx: ContractTransactionResponse
+
+          before(async () => {
+            await tbtc.mint(await mezoAllocator.getAddress(), donationAmount)
+
+            tx = await mezoAllocator.connect(governance).releaseDeposit()
+          })
+
+          it("should not emit DepositReleased event", async () => {
+            await expect(tx).to.not.emit(mezoAllocator, "DepositReleased")
+          })
+
+          it("should transfer tBTC to stBTC contract", async () => {
+            await expect(tx).to.changeTokenBalances(
+              tbtc,
+              [mezoAllocator, stbtc],
+              [-donationAmount, donationAmount],
+            )
+          })
+
+          it("should not call MezoPortal.withdraw function", async () => {
+            await expect(tx).to.not.emit(mezoPortal, "WithdrawFully")
+          })
+        })
+      })
+
       context("when there is a deposit", () => {
-        let tx: ContractTransactionResponse
+        beforeAfterSnapshotWrapper()
+
+        const depositAmount = to1e18(5)
 
         before(async () => {
-          await tbtc.mint(await stbtc.getAddress(), to1e18(5))
+          await tbtc.mint(await stbtc.getAddress(), depositAmount)
           await mezoAllocator.connect(maintainer).allocate()
-          tx = await mezoAllocator.connect(governance).releaseDeposit()
         })
 
-        it("should emit DepositReleased event", async () => {
-          await expect(tx)
-            .to.emit(mezoAllocator, "DepositReleased")
-            .withArgs(1, to1e18(5))
-        })
+        context("when there is no donation", () => {
+          beforeAfterSnapshotWrapper()
+
+          let tx: ContractTransactionResponse
+
+          before(async () => {
+            tx = await mezoAllocator.connect(governance).releaseDeposit()
+          })
+
+          it("should emit DepositReleased event", async () => {
+            await expect(tx)
+              .to.emit(mezoAllocator, "DepositReleased")
+              .withArgs(1, depositAmount)
```

Comment:
I noticed that we do not define the deposit id as a variable in most places. In my opinion, extracting the value to a variable with a good name will improve readability but ofc non-blocking.

Reply:
93a781f0aced7c7a5f2a7c7e0cb3be0a0bf47077